### PR TITLE
[improve][broker][PIP-195] Add topicName and cursorName for ledger metadata of bucket snapshot

### PR DESCRIPTION
**Pull request by coderzc**
PIP: #16763 

### Motivation & Modifications

add topicName and cursorName for BK metadata of bucket snapshot, so we know who is using the ledger.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]
Good catch!

I think it should be one that we missed in the proposal
Could you please also update the proposal according 
to the changes of the Ledger's metadata? It can help
users to troubleshoot issues like why the ledger has not
been deleted.

**Comment by coderzc** [comment]
> Good catch!
> 
> I think it should be one that we missed in the proposal Could you please also update the proposal according to the changes of the Ledger's metadata? It can help users to troubleshoot issues like why the ledger has not been deleted.

Yes, I will update this part of the proposal

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/LedgerMetadataUtils.java [hunk 1]
```diff
@@ -48,7 +48,9 @@ public final class LedgerMetadataUtils {
     private static final String METADATA_PROPERTY_COMPACTEDTO = "pulsar/compactedTo";
     private static final String METADATA_PROPERTY_SCHEMAID = "pulsar/schemaId";
 
-    private static final String METADATA_PROPERTY_DELAYED_INDEX_BUCKETID = "pulsar/delayedIndexBucketId";
+    private static final String METADATA_PROPERTY_DELAYED_INDEX_BUCKET_KEY = "pulsar/delayedIndexBucketKey";
```
Comment (Technoboy-) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/LedgerMetadataUtils.java:51]: Does this change affect getting older values?

Comment (coderzc) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/LedgerMetadataUtils.java:51]: No, we don't read metadata directly, it only used to troubleshoot issues.

**Pull request merged**